### ISSUE TITLE
problem "WITH" on INSERT

### ISSUE THREAD

**Issue by ivanavguston-oviva** [phpmyadmin/sql-parser#353]
it legitimate construction but linter does not allow it

```
 Line 2, col 1 at "WITH": Unexpected keyword.
 Line 2, col 1 at "WITH": Unrecognized statement type.
```
`INSERT ... WITH ... SELECT ... ` https://dev.mysql.com/doc/refman/8.0/en/with.html

**Comment by williamdes** [comment]
Thanks for reporting this !

**Comment by ivanavguston-oviva** [comment]
Hello folks, do you have any update for me?

**Comment by williamdes** [comment]
> Hello folks, do you have any update for me?

Hi !
Actually we recently added the WITH keyword, see #165 and #331 

Could you test the [dev-master](https://packagist.org/packages/phpmyadmin/sql-parser#dev-master) version of the library in case it already improved this ?

Hi @iifawzi 
Would you mind having a look if this is now covered ?

**Comment by iifawzi** [comment]
Hi, I can confirm that it's not covered. Also such form of statements are not covered too. 
```
CREATE TABLE ... WITH ... SELECT ...
REPLACE ... WITH ... SELECT ...
``` 
Examples: 
```SQL 
INSERT INTO table_name WITH cte (col1) AS ( SELECT 1 UNION ALL SELECT 2 ) SELECT col1 FROM cte
INSERT INTO table_name WITH cte (col1) AS ( SELECT 1 UNION ALL SELECT 2 ) SELECT col1 FROM cte ON DUPLICATE KEY UPDATE number = 3
CREATE table table_name AS WITH cte (col1) AS ( SELECT 1 UNION ALL SELECT 2 ) SELECT col1 FROM cte
```
I couldn't think of a test case for the REPLACE statement, anyway other tests added at [iifawzi/sql-parser/support-with-statements](https://github.com/iifawzi/sql-parser/tree/support-with-statements). 

We need to make other statements aware of the keyword. I will find some time to work on it.

**Comment by mauriciofauth** [comment]
Closed by #363
